app/src/main/java/com/douglasbruce/tasky/MainActivity.kt

```diff
@@ -32,40 +25,29 @@ class MainActivity : ComponentActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         val splashScreen = installSplashScreen()
         super.onCreate(savedInstanceState)
-
-        var uiState: MainActivityUiState by mutableStateOf(Loading)
-
-        // Update the uiState
-        lifecycleScope.launch {
-            lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED) {
-                viewModel.uiState.onEach {
-                    uiState = it
-                }.collect()
-            }
-        }
-
         splashScreen.setKeepOnScreenCondition {
-            when (uiState) {
-                Loading -> true
-                is Success -> false
-            }
+            viewModel.isLoading.value
         }
-
+        enableEdgeToEdge()
         setContent {
-            TaskyTheme {
-                val navController: NavHostController = rememberNavController()
-                TaskyNavHost(
-                    navController = navController,
-                    startDestination = if (isUserLoggedIn(uiState)) agendaGraphRoute else loginGraphRoute
-                )
-            }
+            TaskyApp(viewModel)
         }
     }
 }
 
-private fun isUserLoggedIn(
-    uiState: MainActivityUiState,
-): Boolean = when (uiState) {
-    Loading -> false
-    is Success -> uiState.isUserLoggedIn
+@Composable
+fun TaskyApp(viewModel: MainActivityViewModel) {
+    TaskyTheme {
+        val isLoading by viewModel.isLoading.collectAsState()
```

Comment:
Recently the collectAsStateWithLifecycle() dependency got stable, so I'd recommend to use that since it's the safest way to collect flows in Android. For that, you'll need this dependency:
```
implementation "androidx.lifecycle:lifecycle-runtime-compose:2.x.x"
```
Then use it like this:
```
val state by viewModel.state.collectAsStateWithLifecycle()
```
If you have a simple flow like here it shouldn't make a difference, but as soon as you convert a normal `Flow` into a `StateFlow` with `stateIn`, the Flow operators will be executed even when the app is in the background (which you typically don't want when dealing with UI updates).